contrib/automation_tests/test_cases/symbols_tab.py

```diff
@@ -137,45 +146,36 @@ def _verify_all_modules_are_cached(self, modules: List[Module]):
             'All successfully loaded modules are cached. Modules not found in cache: {}'.format(
                 [module.name for module in module_set]))
 
-    def _wait_for_loading_and_collect_errors(self) -> ModulesLoadingResult:
-        assume_loading_complete = 0
-        num_assumptions_to_be_safe = 5
-        error_modules = set()
-        total_time = 0
-
-        while assume_loading_complete < num_assumptions_to_be_safe:
-            start_time = time.time()
-
-            time.sleep(1)
-
-            # Since there is no "loading completed" feedback, give orbit some time to update the status message or
-            # show an error dialog. Then check if any of those is visible.
-            # If not, try a few more times to make sure we didn't just accidentally query the UI while the status
-            # message was being updated.
-            error_module = _find_and_close_error_dialog(self.suite.top_window())
-            status_message = self.find_control('StatusBar', recurse=False).texts()[0]
-            if "Copying debug info file" not in status_message and "Loading symbols" not in status_message:
-                status_message = None
-
-            if error_module is not None:
-                error_modules.add(error_module)
-
-            if not error_module and not status_message:
-                assume_loading_complete += 1
-            else:
-                total_time += time.time() - start_time
-                assume_loading_complete = 0
-        logging.info("Assuming symbol loading has completed. Total time: {time:.2f} seconds".format(
+    def _wait_for_loading_and_measure_time(self) -> float:
```

Comment:
Should this function have a timeout?

Reply:
Yeah, good idea. I've added a rather generous timeout.